R/utilities-data-set.R

```diff
@@ -55,38 +55,64 @@ saveDataSetToPKML <- function(dataSet, filePath) {
   rClr::clrCall(dataRepositoryTask, "SaveDataRepository", dataSet$dataRepository$ref, filePath)
 }
 
-#' Converts a list of DataSet objects to a data.frame
+#' Converts a list of `DataSet` objects to a data.frame
 #'
-#' @param dataSets A list of DataSet objects or a single DataSet
+#' @param dataSets A list of `DataSet` objects or a single `DataSet`
 #'
-#' @return DataSet objects as data.frame with columns name, xValue, yValue, yErrorValues,
-#' xDimension, xUnit, yDimension, yUnit, yErrorType, yErrorUnit, yMolWeight
+#' @return DataSet objects as data.frame with columns name, xValues, yValues, yErrorValues,
+#' xDimension, xUnit, yDimension, yUnit, yErrorType, yErrorUnit, molWeight, lloq,
+#'  and a column for each meta data that is present in any `DataSet`
 #' @export
 dataSetToDataFrame <- function(dataSets) {
   dataSets <- c(dataSets)
   validateIsOfType(dataSets, DataSet)
 
-  name <- unlist(mapply(.makeDataFrameColumn, dataSets, "name"))
-  xUnit <- unlist(mapply(.makeDataFrameColumn, dataSets, "xUnit"))
-  yUnit <- unlist(mapply(.makeDataFrameColumn, dataSets, "yUnit"))
-  yErrorUnit <- as.character(unlist(mapply(.makeDataFrameColumn, dataSets, "yErrorUnit")))
-  xDimension <- unlist(mapply(.makeDataFrameColumn, dataSets, "xDimension"))
-  yDimension <- unlist(mapply(.makeDataFrameColumn, dataSets, "yDimension"))
-  yErrorType <- as.character(unlist(mapply(.makeDataFrameColumn, dataSets, "yErrorType")))
-  yMolWeight <- unlist(mapply(.makeDataFrameColumn, dataSets, "molWeight"))
-  xValue <- unlist(mapply(.makeDataFrameColumn, dataSets, "xValues"))
-  yValue <- unlist(mapply(.makeDataFrameColumn, dataSets, "yValues"))
-  yErrorValues <- unlist(mapply(.makeDataFrameColumn, dataSets, "yErrorValues"))
+  name <- unlist(lapply(dataSets, function(x) {

```

Comment:
mapply and sapply and so produce weird results sometimes (in case of data sets loaded from excel the output was a matrix, in the case of manually created data sets it was a vector...), `lapply` is the only function that _always_ will return a list, so lets use it where possible.

Adding `use.naames = FALSE` to `unlist` speeds up the whole thing and should be used when we do not need a named list.

Reply:
@hannaei FYI